libraries/AP_HAL_PX4/UARTDriver.cpp

```diff
@@ -93,19 +87,13 @@ void PX4UARTDriver::begin(uint32_t b, uint16_t rxS, uint16_t txS)
     /*
       allocate the write buffer
      */
-	if (txS != 0 && txS != _writebuf_size) {
+    if (txS != 0 && txS != _writebuf.get_size()) {
         _initialised = false;
         while (_in_timer) {
             hal.scheduler->delay(1);
         }
-		_writebuf_size = txS;
-		if (_writebuf != NULL) {
-			free(_writebuf);
-		}
-		_writebuf = (uint8_t *)malloc(_writebuf_size+16);
-		_writebuf_head = 0;
-		_writebuf_tail = 0;
-	}
+        _writebuf.set_size(txS);
```

Comment:
I understand that we can't add 16 here, but do we know why the *malloc* had that 16 added? I just want to make sure this wasn't a workaround for an underlying PX4 bug.